addons/website_sale/models/delivery_carrier.py

```diff
@@ -12,3 +16,53 @@ class DeliveryCarrier(models.Model):
         related='product_id.description_sale',
         readonly=False,
     )
+    has_estimated_delivery = fields.Boolean(
+        string='Estimated Delivery',
+        help="Display estimated date to your customer. Editable if range is defined."
+    )
+    estimated_delivery_start = fields.Integer()
+    estimated_delivery_end = fields.Integer()
+    opening_hours = fields.Many2one(
+        string="Opening Hours", comodel_name='resource.calendar', check_company=True
+    )
+
+    def _get_available_days(self):
+        """Return the available days defined on the estimated delivery field based on the calendar.
+
+        :returns: list of all available days of a given range.
+        :rtype: list
+        """
+        self.ensure_one()
+        available_days = []
+        availability_range = self.estimated_delivery_end + self.estimated_delivery_start
+        if self.has_estimated_delivery:
+            # `_attendance_intervals_batch` requires the datetime to be timezoned
+            tz = timezone(self.opening_hours.tz)
+            current_date = fields.Datetime.now()
+            from_datetime = (current_date).astimezone(tz)
+            # add 30 days as buffer to account for unavailable days
+            to_datetime = (
+                current_date + timedelta(days=self.estimated_delivery_end + 30)
+            ).astimezone(tz)
+            availabilities = self.opening_hours._attendance_intervals_batch(
+                from_datetime, to_datetime)[False]._items
+            for availability in availabilities:
+                # availability is a tuple where the first element is the datetime object
+                availability_date = availability[0].date()
+                if availability_date not in available_days:
+                    available_days.append(availability_date)
+                if len(available_days) == availability_range:
+                    break
+        return available_days[self.estimated_delivery_start : self.estimated_delivery_end + 1]
+
+    def _get_first_available_delivery_date(self):
+        """Return the first available day for delivery called when `estimated_delivery_start`
+        and `estimated_delivery_end` are equal so no need to return the full range of available
+        days.
```

Comment:
Im not convinced the second part of the docstring is useful without knowing the entire task. wdyt?
```suggestion
        """Return the first available day for delivery.        
```